index.js

```diff
@@ -48,25 +57,34 @@ exports = module.exports = (function () {
     function getHarness(opts) {
         if (!opts) opts = {};
         opts.autoclose = !canEmitExit;
-        if (!harness) harness = createExitHarness(opts);
+        if (!harness) harness = createExitHarness(opts, wait);
         return harness;
     }
 })();
 
-function createExitHarness(conf) {
+function createExitHarness(conf, wait) {
     if (!conf) conf = {};
     var harness = createHarness({
         autoclose: defined(conf.autoclose, false)
     });
+    var running = false;
+    var ended = false;
 
-    var stream = harness.createStream({ objectMode: conf.objectMode });
-    var es = stream.pipe(conf.stream || createDefaultStream());
-    if (canEmitExit) {
-        es.on('error', function (err) { harness._exitCode = 1; });
-    }
+    harness.run = function () {
+        if (running) return;
+        running = true;
+        var stream = harness.createStream({ objectMode: conf.objectMode });
+        var es = stream.pipe(conf.stream || createDefaultStream());
+        if (canEmitExit) {
+            es.on('error', function (err) { harness._exitCode = 1; });
+        }
+        stream.on('end', function () { ended = true; });
+    };
 
-    var ended = false;
-    stream.on('end', function () { ended = true; });
+
+    if (!wait) {
+        harness.run();
+    }
```

Comment:
maybe when `wait` is falsy, `harness.run` shouldn't exist? iow, maybe this should be `if (wait) { harness.run = run; } else { run(); }`?

Reply:
could be done, i would have to update `lazyload.run` a little bit to:

```js
lazyLoad.run = function () {
    var run = getHarness().run;
    
    if (run) {
        run();
    }
};
```

Reply:
or `lazyload.run = getHarness().run || function () {};`?

Reply:
unfortunately that logic is called on `require('tape')`, so we will not be able to call `tape.wait()` before. it will also make it impossible to override the default stream.